src/activity_actor.cpp

```diff
@@ -2208,6 +2208,47 @@ std::unique_ptr<activity_actor> stash_activity_actor::deserialize( JsonIn &jsin
     return actor.clone();
 }
 
+void move_furniture_activity_actor::start( player_activity &act, Character & )
+{
+    int moves = g->grabbed_furn_move_time( dp );
+    act.moves_left = moves;
+    act.moves_total = moves;
+}
+
+void move_furniture_activity_actor::finish( player_activity &act, Character &who )
+{
+    if( !g->grabbed_furn_move( dp ) ) {
+        g->walk_move( who.pos() + dp, via_ramp, true );
+    }
+    act.set_to_null();
+}
+
+void move_furniture_activity_actor::canceled( player_activity &, Character &who )
+{
+    add_msg( m_warning, _( "You let go of the grabbed object." ) );
+    get_avatar().grab( object_type::NONE );
```

Comment:
```suggestion
    who.grab( object_type::NONE );
```

Reply:
grab doesn't exist in Character.  I don't know that it makes sense to move it there.

Reply:
Then you should ditch the `who` argument.